test/e2e/node/kubelet.go

```diff
@@ -309,52 +304,52 @@ var _ = SIGDescribe("kubelet", func() {
-				err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, totalPods, time.Second*30)
-				framework.ExpectNoError(err)
-				if resourceMonitor != nil {
-					resourceMonitor.LogLatest()
-				}
+		ginkgo.It("should let Kubelet delete 10 pods per node with in 1 minute", func() {
+			totalPods := 10 * numNodes
+			ginkgo.By(fmt.Sprintf("Creating a RC of %d pods and wait until all pods of this RC are running", totalPods))
+			rcName := fmt.Sprintf("cleanup%d-%s", totalPods, string(uuid.NewUUID()))
 
-				ginkgo.By("Deleting the RC")
-				framework.DeleteRCAndWaitForGC(f.ClientSet, f.Namespace.Name, rcName)
-				// Check that the pods really are gone by querying /runningpods on the
-				// node. The /runningpods handler checks the container runtime (or its
-				// cache) and  returns a list of running pods. Some possible causes of
-				// failures are:
-				//   - kubelet deadlock
-				//   - a bug in graceful termination (if it is enabled)
-				//   - docker slow to delete pods (or resource problems causing slowness)
-				start := time.Now()
-				err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, 0, itArg.timeout)
-				framework.ExpectNoError(err)
-				e2elog.Logf("Deleting %d pods on %d nodes completed in %v after the RC was deleted", totalPods, len(nodeNames),
-					time.Since(start))
-				if resourceMonitor != nil {
-					resourceMonitor.LogCPUSummary()
-				}
+			label := labels.SelectorFromSet(labels.Set(map[string]string{"name": rcName}))
+			ps, err := testutils.NewPodStore(f.ClientSet, f.Namespace.Name, label, fields.Everything())
+			framework.ExpectNoError(err)
+			interval := 10 * time.Second
+
+			err = framework.RunRC(testutils.RCConfig{
+				Client:       f.ClientSet,
+				Name:         rcName,
+				Namespace:    f.Namespace.Name,
+				Image:        imageutils.GetPauseImageName(),
+				Replicas:     totalPods,
+				NodeSelector: nodeLabels,
 			})
-		}
+			framework.ExpectNoError(err)
+			// Perform a sanity check so that we know all desired pods are
+			// running on the nodes according to kubelet. The timeout is set to
+			// only 30 seconds here because framework.RunRC already waited for all pods to
+			// transition to the running status.
+			err = e2epod.WaitForPodsGone(ps, interval, 10*time.Second)
+			framework.ExpectError(err)
+			if resourceMonitor != nil {
+				resourceMonitor.LogLatest()
+			}
+
+			ginkgo.By("Deleting the RC")
+			framework.DeleteRCAndWaitForGC(f.ClientSet, f.Namespace.Name, rcName)
+			// Check that the pods really are gone by querying /runningpods on the
+			// node. The /runningpods handler checks the container runtime (or its
+			// cache) and  returns a list of running pods. Some possible causes of
+			// failures are:
+			//   - kubelet deadlock
+			//   - a bug in graceful termination (if it is enabled)
+			//   - docker slow to delete pods (or resource problems causing slowness)
+			start := time.Now()
+			err = e2epod.WaitForPodsGone(ps, interval, 10*time.Second)
```

Comment:
again, I think this wait is redundant and should be removed

this is already called in `DeleteResourceAndWaitForGC` (which is called by `DeleteRCAndWaitForGC`)

https://github.com/kubernetes/kubernetes/blob/bc1d8c6d61078269540f807248a455cba09e672c/test/e2e/framework/util.go#L2279